app/views/auth_key_pair_cloud/_form.html.haml

```diff
@@ -1,44 +1,53 @@
-#main_div
-  - url = url_for(:action => 'form_field_changed', :id => @key_pair.id || "new")
+%h3
+  = _('Basic Information')
+  %br
+  %br
 
-  = render :partial => "layouts/flash_msg"
-
-  %h3
-    = _('Basic Information')
-  .form-horizontal
-    .form-group
-      %label.col-md-2.control-label
+.form-horizontal{:id => "start_form_div", :style => "display:none"}
+  %form#form_div{:name           => "angularForm",
+                 'ng-controller' => "keyPairCloudFormController",
+                 'ng-show'       => "afterGet",
+                 :novalidate     => true}
+    = render :partial => "layouts/flash_msg"
+    .form-group{"ng-class" => "{'has-error': angularForm.name.$invalid}"}
+      %label.col-md-2.control-label{"for" => "name"}
         = _('Name')
       .col-md-8
-        = text_field_tag("name",
-                          @edit[:new][:name],
-                          :maxlength => 50,
-                          :class => "form-control",
-                          "data-miq_focus" => true,
-                          "data-miq_observe" => {:interval => '.5', :url => url}.to_json)
+        %input.form-control{:type            => "text",
+                            :name            => "name",
+                            "id"             => "name",
```

Comment:
do we need to mix string and symbol keys?

maybe for ng-mode, althoug I thing there's `_` to `-` conversion

surely not for `id`